### ISSUE TITLE
[a11y]: Text preceding file uploader needs programmatic indicators

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#11196]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

https://react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader

### React version

https://react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader

### Automated testing tool and ruleset

inspect and manual testing

### Description

Issue identified during the recent Carbon Accessibility Guild call. 
<img width="455" alt="image" src="https://user-images.githubusercontent.com/14298245/162796061-07fa766a-24bd-488a-8190-757cdca21c46.png">

The text preceding the Add file button lacks meaningful structure.
The description (2 in above image) ("Max file size is 500kb...") that comes underneath the Upload files heading should be associated with the Add file button (3 in above image) using aria-describedby

![image](https://user-images.githubusercontent.com/14298245/162796606-df67cae7-b673-498a-941c-9c35062ae919.png)

`<h3 class="cds--file--label">Upload files</h3>`
`<p id="description" class="cds--label-description">Max file size is 500mb. Only .jpg files are supported.</p>`
`<label tabindex="0" class="cds--btn cds--btn--primary cds--btn--md" for="id1" aria-describedby="description"><span role="button" aria-disabled="false">Add file</span></label>`

![image](https://user-images.githubusercontent.com/14298245/162799730-bbb6feb0-b07e-40a4-a5fe-4065141eaa11.png)

As well, arguably the "Upload files" text should be a heading, not just a paragraph. Design has already agreed to call this a 'heading' in the anatomy, not a label (since it labels nothing; the button has its own label). If it's possible to have it set as a relative heading, that would be best. If a designated heading is too prescriptive, this can stay at just a paragraph without significant user impact. 

### WCAG 2.1 Violation

1.3.1 Info and relationships

### CodeSandbox example

https://react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader

### Steps to reproduce

go to storybook.
confirm the description and

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by mbgower** [comment]
@joshblack @sstrubberg Just following up on this.
Once Carbon makes a _decision_ on how/if to surface this Description accessibly, then I need to capture that decision in the Accessibility tab (either that Carbon provides it and how it is achieved, mentioned in the Development section -- preferred OR that designers need to annotate it , mentioned in the Design Recommendations).
So please LMK when a tactical decision is made, even if it hasn't been implemented yet, and I will update.
